### PR TITLE
feat: add support for getBooleanEvaluation with default value

### PR DESCRIPTION
**Pull request by liran2000**
add support for getBooleanEvaluation with default value

@weihao-statsig @xinlili-statsig

### CODE REVIEW

**Comment by weihao-statsig** [comment]
Thanks for changing this. Let me know if you find any other issues

File: providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java [hunk 1]
```diff
@@ -82,11 +85,21 @@ public Metadata getMetadata() {
...
         StatsigUser user = ContextTransformer.transform(ctx);
         Boolean evaluatedValue = defaultValue;
-        try {
+        Value featureConfigValue = ctx.getValue(FEATURE_CONFIG_KEY);
+        String reason = null;
+        if (featureConfigValue == null) {
+            APIFeatureGate featureGate = Statsig.getFeatureGate(user, key);
+            if (featureGate.getReason() != null && assumeFailure(featureGate)) {
+                reason = featureGate.getReason().getReason();
+            } else {
```
Comment (toddbaert) [inline providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java]: I don't quite understand this. Could you perhaps add a comment?

Comment (toddbaert) [reply providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java]: I think the thing that's confusing me is it seems we are only setting the reason if the reason is a failure?

Comment (liran2000) [reply providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java]: I updated it a bit and added comment here and on the _assumeFailure()_ method.

**Review by toddbaert — APPROVED** [review]

File: providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java (hunk 1 above)
Comment (toddbaert) [reply providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java]: Thanks!

**Pull request merged**